src/platform/web/ui/general/theming.js

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export function getStyleSheetElements() {
+    const styleSheets = Object.create(null);
+    for (const link of document.getElementsByTagName("link")) {
+        const match = link.href.match(/^.*\/themes\/(.*)\/.*\.css$/);
+        if (match) {
+            styleSheets[match[1]] = link;
+        }
+    }
+    return styleSheets;
+}
+
+export function setTheme(name, styleSheetElements) {
+    if (!styleSheetElements) {
+        styleSheetElements = getStyleSheetElements();
+    }
+
+    if (!(name in styleSheetElements)) {
+        return;
+    }
+
+    styleSheetElements[name].disabled = false;
+
+    return new Promise(resolve => {
+        const switchTheme = function() {
+            styleSheetElements[name].disabled = false; // Poor man's race condition precaution
```

Comment:
could you explain the race a bit more here?

Reply:
I shamelessly copied that part from [this](https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/theme.js#L208) spot in matrix-react-sdk. There's a longer comment there with a reference to a GitHub issue. From what I understand, since `switchTheme` is triggered asynchronously via the `onload` handler, it could happen that another theme change request came in shortly after which then leads to a race for setting `disabled`. According to the mentioned GitHub heading this only affected Chrome.